### PR TITLE
Bug 1797766:  Resource Requirements" specDescriptor fields - CPU and Memory injects empty string YAML editor 

### PR DESCRIPTION
**Pull request by cyril-ui-developer**
/assign @TheRealJon
/cc @spadgett

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/resource-requirements.tsx [hunk 1]
```diff
@@ -79,13 +79,12 @@ export const ResourceRequirements: React.FC<ResourceRequirementsProps> = (props)
 export const ResourceRequirementsModal = withHandlePromise(
   (props: ResourceRequirementsModalProps) => {
     const { obj, path, type, model } = props;
-    const [cpu, setCPU] = React.useState<string>(_.get(obj.spec, `${path}.${type}.cpu`, ''));
-    const [memory, setMemory] = React.useState<string>(
-      _.get(obj.spec, `${path}.${type}.memory`, ''),
-    );
+    const [cpu, setCPU] = React.useState<string>(_.get(obj.spec, `${path}.${type}.cpu`));
```
Comment (benjaminapetersen) [inline frontend/packages/operator-lifecycle-manager/src/components/descriptors/spec/resource-requirements.tsx:82]: Is this a candidate for the typescript syntax instead? `obj.spec?.path?.type?.cpu`

**Comment by spadgett** [comment]
@TheRealJon PTAL
@cyril-ui-developer Sorry, needs rebase :/

**Comment by cyril-ui-developer** [comment]
@spadgett Rebased. Thanks

**Comment by cyril-ui-developer** [comment]
@spadgett Just rebase this PR -https://github.com/openshift/console/pull/5525 again. Please take a look when you have a chance. Thanks

**Review by TheRealJon — CHANGES_REQUESTED** [review]
This is untested, but I would think something like this would work:

```tsx
const submit = (e) => {
  e.preventDefault();
  const newObj = _.set(_.clone(obj), `spec.${path}`, {
    ...(memory && { memory }),
    ...(cpu && { cpu }),
    ...(storage && { 'ephemeral-storage': storage }),
  });
  return props.handlePromise(k8sUpdate(model, newObj), props.close);
};
```

**Review by cyril-ui-developer — COMMENTED** [review]
> This is untested, but I would think something like this would work:
> 
> ```tsx
> const submit = (e) => {
>   e.preventDefault();
>   const newObj =
>     cpu || memory || storage
>       ? _.set(_.cloneDeep(obj), `spec.${path}.${type}`, {
>           ...(cpu && { cpu }),
>           ...(memory && { memory }),
>           ...(storage && { 'ephemeral-storage': storage }),
>         })
>       : obj;
>   return props.handlePromise(k8sUpdate(model, newObj), props.close);
> };
> ```

@TheRealJon Looks cleaner! Works with small tweak. 
PTAL.  Thanks

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**